backend/user/views.py

```diff
@@ -21,3 +31,95 @@ class UserView(generics.RetrieveUpdateAPIView):
 
     def get_object(self):
         return self.request.user
+
+
+class NotificationTokenView(viewsets.ModelViewSet):
+    """
+    get:
+    Return notification tokens of user.
+    """
+
+    permission_classes = [IsAuthenticated]
+    serializer_class = NotificationTokenSerializer
+
+    def get_queryset(self):
+        return NotificationToken.objects.filter(user=self.request.user)
+
+
+class NotificationSettingView(viewsets.ModelViewSet):
+    """
+    get:
+    Return notification settings of user.
+
+    post:
+    Creates/updates new notification setting of user for a specific service.
+
+    check:
+    Checks if user wants notification for specified serice.
+    """
+
+    permission_classes = [IsAuthenticated]
+    serializer_class = NotificationSettingSerializer
+
+    def get_queryset(self):
+        return NotificationSetting.objects.filter(token__user=self.request.user)
+
+    @action(detail=True, methods=["get"])
+    def check(self, request, pk=None):
+        """
+        Returns whether the user wants notification for specified service.
+        :param pk: service name
+        """
+        # style
+        pk = pk.upper().replace("-", "_")
```

Comment:
i think that this probably won't be necessary. reason being is that the iOS team won't know about this niche thing. although it does look nicer, funcitonality-wise i think it might be best to just leave it as PENN_MOBILE in the url